StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/PropertySummaryDocumentationAnalyzer.cs

```diff
@@ -225,51 +228,85 @@ private static void AnalyzeSummaryElement(SyntaxNodeAnalysisContext context, Xml
+                    // Both getter and setter exist, but only setter is visible.
+                    if (!prefixIsSets)
+                    {
+                        if (prefixIsGetsOrSets)
+                        {
+                            ReportSa1624(context, diagnosticLocation, diagnosticProperties, accessor: "set", expectedStartingText: startingTextSets, startingTextToRemove: startingTextGetsOrSets);
+                        }
+                        else
+                        {
+                            ReportSa1623(context, diagnosticLocation, diagnosticProperties, text, expectedStartingText: startingTextSets, unexpectedStartingText1: startingTextGets);
+                        }
+                    }
                 }
-                else if (!text.StartsWith(startingTextSets, StringComparison.OrdinalIgnoreCase))
+                else
                 {
-                    diagnosticProperties.Add(ExpectedTextKey, startingTextSets);
-                    context.ReportDiagnostic(Diagnostic.Create(SA1623Descriptor, diagnosticLocation, diagnosticProperties.ToImmutable(), startingTextSets));
+                    // Setter exists and is visible. Getter does not exist.
+                    if (!prefixIsSets)
+                    {
+                        ReportSa1623(context, diagnosticLocation, diagnosticProperties, text, expectedStartingText: startingTextSets, unexpectedStartingText1: startingTextGetsOrSets, unexpectedStartingText2: startingTextGets);
+                    }
                 }
             }
         }
+
+        private static void ReportSa1623(SyntaxNodeAnalysisContext context, Location diagnosticLocation, ImmutableDictionary<string, string>.Builder diagnosticProperties, string text, string expectedStartingText, string unexpectedStartingText1, string unexpectedStartingText2 = null)
+        {
+            diagnosticProperties.Add(ExpectedTextKey, expectedStartingText);
+
+            if (text.StartsWith(unexpectedStartingText1, StringComparison.OrdinalIgnoreCase))
+            {
+                diagnosticProperties.Add(TextToRemoveKey, text.Substring(0, unexpectedStartingText1.Length));
+            }
+            else if ((unexpectedStartingText2 != null) && text.StartsWith(unexpectedStartingText2, StringComparison.OrdinalIgnoreCase))
+            {
+                diagnosticProperties.Add(TextToRemoveKey, text.Substring(0, unexpectedStartingText2.Length));
+            }
+
+            context.ReportDiagnostic(Diagnostic.Create(SA1623Descriptor, diagnosticLocation, diagnosticProperties.ToImmutable(), expectedStartingText));
+        }
+
+        private static void ReportSa1624(SyntaxNodeAnalysisContext context, Location diagnosticLocation, ImmutableDictionary<string, string>.Builder diagnosticProperties, string accessor, string expectedStartingText, string startingTextToRemove)
```

Comment:
I would prefer `ReportSA1624` here, to more clearly preserve the link with the identifier that is reported.